borg/crypto.pyx

```diff
@@ -124,7 +143,7 @@ cdef class AES:
         if not out:
             raise MemoryError
         try:
-            if not EVP_DecryptUpdate(&self.ctx, out, &outl, data, inl):
+            if not EVP_DecryptUpdate(&self.ctx, out, &outl, <const unsigned char*> data_buf.buf, inl):
```

Comment:
This is the code as generated by this commit for that function call.

`__pyx_v_data_buf` is the buffer as created by ro_buffer, and PyObject_GetBuffer never copies (for any type it should either provide direct access to the underlying memory of the object as specified by the flags etc. or it should raise an exception).